src/hip/hip_build_utils.cpp

```diff
@@ -137,6 +137,26 @@ boost::filesystem::path HipBuild(boost::optional<TmpDir>& tmp_dir,
             MIOPEN_LOG_E("failed to find *.hsaco in " << hsaco->path().string());
         }
 
+        return hsaco->path();
+    }
+#elif OFFLOADBUNDLER_BIN
+    if(IsClangXXCompiler())
+    {
+        // call clang-offload-bundler
+        tmp_dir->Execute(OFFLOADBUNDLER_BIN,
+                         "-type=o -targets=hip-amdgcn-amd-amdhsa-" + dev_name + " -inputs=" +
+                             bin_file.string() + " -unbundle");
+
+        auto hsaco =
+            std::find_if(boost::filesystem::directory_iterator{tmp_dir->path},
+                         {},
+                         [](auto entry) { return (entry.path().extension() == ".hsaco"); });
```

Comment:
If you do not specify the output file name by -o when doing device-only compilation with HIP-Clang, the default output file name is like *.hipfb.

Better test the patch locally to make sure it works.

Reply:
I think I am picking up extractkernels still. Let me go back and fix this.

Reply:
@yxsamliu are there any other differences I need to be aware of?